scheds/rust/scx_layered/src/main.rs

```diff
@@ -1099,6 +1111,111 @@ impl Layer {
+}
+
+impl OpenMetricsStats {
+    fn new() -> OpenMetricsStats {
+        let mut metrics = OpenMetricsStats {
+            registry: <Registry>::default(),
+            ..Default::default()
+        };
+        // Helper macro to reduce on some of the boilerplate:
+        // $i: The identifier of the metric to register
+        // $help: The Help text associated with the metric
+        macro_rules! register {
+            ($i:ident, $help:expr) => {
+                metrics
+                    .registry
+                    .register(stringify!($i), $help, metrics.$i.clone())
+            };
+        }
+        register!(total, "Total scheduling events in the period");
+        register!(local, "% that got scheduled directly into an idle CPU");
+        register!(
+            open_idle,
+            "% of open layer tasks scheduled into occupied idle CPUs"
+        );
+        register!(
+            affn_viol,
+            "% which violated configured policies due to CPU affinity restrictions"
+        );
+        register!(tctx_err, "Failures to free task contexts");
+        register!(
+            proc_ms,
+            "CPU time this binary has consumed during the period"
+        );
+        register!(busy, "CPU busy % (100% means all CPUs were fully occupied)");
+        register!(
+            util,
+            "CPU utilization % (100% means one CPU was fully occupied)"
+        );
+        register!(load, "Sum of weight * duty_cycle for all tasks");
+        register!(
+            layer_util,
+            "CPU utilization of the layer (100% means one CPU was fully occupied)"
+        );
+        register!(
+            layer_util_frac,
+            "Fraction of total CPU utilization consumed by the layer"
+        );
+        register!(
+            layer_load,
+            "Sum of weight * duty_cycle for tasks in the layer"
+        );
+        register!(
+            layer_load_frac,
+            "Fraction of total load consumed by the layer"
+        );
+        register!(layer_tasks, "Number of tasks in the layer");
+        register!(layer_total, "Number of scheduling events in the layer");
+        register!(
+            layer_local,
+            "% of scheduling events directly into an idle CPU"
+        );
+        register!(
+            layer_open_idle,
+            "% of scheduling events into idle CPUs occupied by other layers"
+        );
+        register!(
+            layer_preempt,
+            "% of scheduling events that preempted other tasks"
+        );
+        register!(
+            layer_affn_viol,
+            "% of scheduling events that violated configured policies due to CPU affinity restrictions"
+        );
+        register!(layer_cur_nr_cpus, "Current # of CPUs assigned to the layer");
+        register!(layer_min_nr_cpus, "Minimum # of CPUs assigned to the layer");
+        register!(layer_max_nr_cpus, "Maximum # of CPUs assigned to the layer");
```

Comment:
This is a bit of an eyesore. Maybe they'll have declarative syntax one day ala clap.

Reply:
Yeah, I tried to reduce this with the macro, but agreed - having it on the struct definition would be nicer